internal/file/zip_file_manifest.go

```diff
@@ -48,7 +48,7 @@ func (z ZipFileManifest) GlobMatch(patterns ...string) []string {
 			// so that glob logic is consistent inside and outside of ZIP archives
 			normalizedEntry := normalizeZipEntryName(entry)
 
-			if GlobMatch(pattern, normalizedEntry) {
+			if GlobMatch(strings.ToLower(pattern), normalizedEntry) {
```

Comment:
@coheigea just one question for this PR - this changes every pattern that uses `GlobMatch` to be case insensitive by lowercasing the inputs. 

Since the purpose of this PR is around `licenses`, what do you think about sanitizing the input of `patterns` where this function is used surrounding license detection. I can do some digging in the code to see what this looks like.

cc @wagoodman if he has any other comments about if this change has any unintended consequences

I see usages of this function under `syft/pkg/cataloger/java`:
```
discoverMainPackage
discoverPkgsFromAllMavenFiles
discoverPkgsFromZip
getLicenseFromFileInArchive
guessMainPackageNameAndVersionFromPomInfo
```

Is it ultimately these calses you want to call `strings.ToLower` on:
https://github.com/anchore/syft/blob/234ce4e1f3b995013551229105a450ed29a6112a/syft/pkg/cataloger/java/archive_parser.go#L341-L347

Reply:
@spiffcs I changed the code to add a parameter to control whether or not case insensitive matching is done. It's false for all the use-cases apart from when I'm matching against a license file.